File: app/src/main/java/vasuki/istanpdf/domain/DocxToPdf.java
package vasuki.istanpdf.domain;

import android.net.Uri;

import java.io.File;

import vasuki.istanpdf.data.DocxEngine;

public final class DocxToPdf {
    private final DocxEngine docxEngine;

    public DocxToPdf(DocxEngine docxEngine) {
        this.docxEngine = docxEngine;
    }

    public File execute(Uri docxUri) throws Exception {
        if (docxUri == null) {
            throw new IllegalArgumentException("DOCX URI is required");
        }
        return docxEngine.exportToPdf(docxUri);
    }
}

File: app/src/main/kotlin/vasuki/istanpdf/domain/DocxToPdf.kt
package vasuki.istanpdf.domain

import android.net.Uri
import vasuki.istanpdf.data.DocxEngine
import java.io.File

class DocxToPdf(private val docxEngine: DocxEngine) {
    @Throws(Exception::class)
    fun execute(docxUri: Uri): File {
        return docxEngine.exportToPdf(docxUri)
    }
}
